Integrations/integration-symantec-DLP-incident-reportintg-update.yml

```diff
@@ -0,0 +1,336 @@
+    session.verify = False
+    cache = SqliteCache(timeout=None)
+    transport = Transport(session=session, cache=cache)
+
+    client = Client(wsdl=wsdl, transport=transport)
+
+    def  get_incident_id(dtobj):
+
+        if type(dtobj) is datetime:
+            return client.service.incidentList(savedReportId=SAVED_REPORT_ID, incidentCreationDateLaterThan=dtobj)
+        else:
+            return_error('excpected datetime object')
+
+
+
+    def incidentDetail(_id):
+
+        return client.service.incidentDetail(incidentId=_id,includeViolations=True,includeHistory=True)
+
+
+    def incidentBinaries(args):
+
+        return client.service.incidentBinaries(**args)
+
+
+    def myconverter(o):
+        if isinstance(o, datetime):
+            return o.__str__()
+
+
+    def fetchincidents():
+        last_run=demisto.getLastRun()
+        if 'lt_time' not in last_run:
+            fetch_time,_ = parse_date_range(FETCH_TIME)
+            last_run['lt_time'] = fetch_time
+        else:
+            last_run['lt_time']= datetime.strptime(last_run['lt_time'],'%d/%m/%Y %H:%M:%S.%f')
+
+        incidentids=get_incident_id(last_run['lt_time'])
+        incidents=[]
+        if len(incidentids['incidentId']) > 0:
+            for incident1 in incidentids['incidentId']:
+                inc=incidentDetail(str(incident1))
+                serialized_obj = helpers.serialize_object(inc[0])
+                incidents.append({
+                    'name':"#"+str(incident1) + " DLP "+ inc[0]['incident']['violatedPolicyRule'][0]['ruleName']+ "RULE has been violated",
+                    'details': json.dumps(serialized_obj,default = myconverter),
+                    'rawJSON': json.dumps(serialized_obj,default = myconverter),
+                    'severity' : SEVERITY[inc[0]['incident']['severity']]
+                })
+                if inc[0]['incident']['incidentCreationDate'].isoformat() > last_run['lt_time'].isoformat():
+                    demisto.setLastRun({'lt_time':inc[0]['incident']['incidentCreationDate'].strftime("%d/%m/%Y %H:%M:%S.%f")})
+        demisto.incidents(incidents)
+
+
+    def updateIncidents():
+        args = demisto.args()
+        inc_id = args['incidentId']
+        args.pop('incidentId')
+        if args['status'] == 'New':
+            args['status'] = 'incident.status.New'
+        return client.service.updateIncidents(updateBatch={'batchId':str(random.randint(11111,99999)),'incidentId':inc_id,'incidentAttributes':args})
+
+
+
+    if demisto.command() == 'fetch-incidents':
+        fetchincidents()
+
+    elif demisto.command() =='test-module':
+        try:
+            inc_list = get_incident_id(datetime.now())
+            demisto.results("ok")
+        except:
+            return_error("Invalid instance details")
+
+    elif demisto.command() == 'sdlp-incident-details':
+        inc_details = incidentDetail( demisto.args()['incidentId'])
```

Comment:
In the `if-else` block there should be only function calling, so all the logic should be in these functions